src/Server.cpp

```diff
@@ -852,33 +950,97 @@ void handleCompleteRequestHeader(Server& server, int clientFd, Connection& conne
 	}
 
 	server.findTargetResource(connection, connection.m_request);
+	if (isCGIRequested(connection)) {
+		connection.m_request.hasCGI = true;
+		ProcessOps processOps;
+		CGIHandler cgiHandler(connection, processOps);
+		if (connection.m_request.httpStatus == StatusInternalServerError) {
+			connection.m_status = Connection::BuildResponse;
+			server.modifyEvent(clientFd, EPOLLOUT);
+			return;
+		}
+		cgiHandler.execute(connection.m_request, connection.location, processOps, server.getEpollWrapper().getEpollFd(),
+			server.getConnections());
+		connection.m_pipeToCGIWriteEnd = cgiHandler.getPipeInWriteEnd();
+		connection.m_pipeFromCGIReadEnd = cgiHandler.getPipeOutReadEnd();
+		connection.m_cgiPid = cgiHandler.getCGIPid();
```

Comment:
Since you pass the whole connection you can save a refrence to it's fds and pid variables. 
Then you don't need to pull them out (since they are the same anyway)

Reply:
Valid, done